src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/CombinationController.php

```diff
@@ -56,15 +88,47 @@ public function getListAction(int $productId, Request $request): JsonResponse
             $offset ?? null
         ));
 
-        return $this->json($this->formatResponse($combinationsList));
+        return $this->json($this->formatListResponse($combinationsList));
+    }
+
+    /**
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))")
+     *
+     * @param int $combinationId
+     * @param Request $request
+     *
+     * @return JsonResponse
+     */
+    public function updateCombinationFromListingAction(int $combinationId, Request $request): JsonResponse
+    {
+        $form = $this->getCombinationItemFormBuilder()->getFormFor($combinationId, [], [
+            'method' => Request::METHOD_PATCH,
+        ]);
+        $form->handleRequest($request);
+
+        try {
+            $result = $this->getCombinationItemFormHandler()->handleFor($combinationId, $form);
+
+            if (!$result->isValid()) {
+                //@todo: handle errors
+                return $this->json(['errors' => $form->getErrors(true)], Response::HTTP_BAD_REQUEST);
```

Comment:
We have a `getFormErrorsForJS` in `FrameworkBundleAdminController` that I think you can use here

Reply:
I used it in `ImageController::addImage` Although it's not really used by the component it, at least, can help understand what went wrong for developers

But I guess the JS component could display the error messages in a growl error even if they are sometimes too technical for a merchant